server/src/main/java/org/candlepin/auth/KeycloakAuth.java

```diff
@@ -104,20 +119,28 @@ public Principal getPrincipal(HttpRequest httpRequest) {
                     return null;
                 }
                 else {
-                    Base64 base64Url = new Base64(true);
-                    String[] splitstring = authArray[1].split("\\.");
-                    String base64EncodedBody = splitstring[1];
-                    String body = new String(base64Url.decode(base64EncodedBody));
-                    JSONObject obj = new JSONObject(body);
-                    String tokenType = obj.getString("typ");
-
-                    if (tokenType.equals("Refresh")) {
-                        //Offline Token authentication
-                        handleRefreshToken(httpRequest, auth);
+                    try {
+                        Base64 base64Url = new Base64(true);
+                        String[] splitstring = authArray[1].split("\\.");
+                        String base64EncodedBody = splitstring[1];
+                        String body = new String(base64Url.decode(base64EncodedBody));
+                        ObjectMapper objectMapper = new ObjectMapper();
+                        JsonNode rootNode = objectMapper.readTree(body);
```

Comment:
Digging into this a bit more, we don't need to be manually parsing these JSON structures. The Keycloak library provides DTOs which are annotated for JSON parsers like Jackson. In this particular case, we should be injecting the ObjectMapper (not creating it new here) and then using it to pull the appropriate request object from Keycloak's own representation package (see: JsonWebToken).